File: Python/Lab/Palindrome.py
num = int(input("Enter a number: "))
temp = num
sum = 0
while temp > 0:
    rem = temp%10
    sum = (sum*10) + rem
    temp = temp//10   # use floor division instead of just plain division
if(sum == num):
    print("Palindrome Number")
else:
    print("Not Palindrome Number")

File: C/Lab/palindrome.c
#include <stdio.h>
#include <string.h>

int main()
{
    char str[1000];
    printf("Enter a string: ");
    gets(str);
    int length = strlen(str);
    int isPalindrome = 1;
    for(int i = 0; i < length/2; i++){
        if(str[i] != str[length - i -1]){
            isPalindrome = 0;
            break;
        }
    }
    if(isPalindrome == 1){
        printf("%s is a palindrome string\n", str);
    } else{
        printf("%s is not a palindrome string\n", str);
    }
    return 0;
}